src/ngx_http_lua_ssl_session_fetchby.c

```diff
@@ -181,6 +181,9 @@ ngx_http_lua_ssl_sess_fetch_handler(ngx_ssl_conn_t *ssl_conn,
 #endif
     u_char *id, int len, int *copy)
 {
+#if defined(TLS1_3_VERSION)
```

Comment:
`#ifdef TLS_1_3_VERSION` is enough here?

Or actually maybe `#if defined(NGX_SSL_TLSv1_3) && defined(TLS1_3_VERSION)` for older Nginx versions linked with newer OpenSSL versions? Not sure from the top of my head which Nginx core those directives were first supported with.

Reply:
@thibaultcha
Checking `NGX_SSL_TLSv1_3` is fine, since if the Nginx doesn't support TLS 1.3 will not use TLS 1.3.

However, I think checking `TLS1_3_VERSION` is enough.
1. `SSL_version()` is inexpensive operation.
2. If you are using an old Nginx, you are probably also using an old lua-nginx-module/OpenResty which doesn't have this commit. And it is quite rare to build an old Nginx with a new OpenSSL.
3. personally speaking, if we can avoid it, it is not a good idea to add additional code to make legacy system happy. If people want to build Nginx in a new Linux distribution, they should build a new one or provide old version of OpenSSL themselves. An old Nginx with a new OpenSSL is not tested well and should be avoided.

Reply:
Yeah, i just meant to switch to the `#ifdef` notation. `NGX_SSL_TLSv1_3` isn’t very helpful anyway, albeit still safer I would argue.

Reply:
@thibaultcha 
Done via the new commit.